src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,595 @@
+	}
+}
+
+//
+// The following block of code was taken from
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/monit.go#L550
+
+// helper function to parse comma separated tags string
+func parseTags(itags string) []string {
+	var tags []string
+	for _, tag := range strings.Split(itags, ",") {
+		tags = append(tags, strings.Trim(tag, " "))
+	}
+	return tags
+}
+
+//
+// The following block of code was taken from (with few changes done)
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/monit.go#L559
+
+// helper function to parse given time-range separated by '-'
+func parseTimes() []int64 {
+	var times []int64
+
+	//additional line added in the adopted code for date support.
+	if strings.Contains(trange, "--") {
+		tms := strings.Split(trange, "--")
+
+		startTime, err := time.Parse(time.RFC3339, tms[0])
+		if err != nil {
+			log.Printf("Unable to parse time, error: %s", err)
+		}
+
+		endTime, err := time.Parse(time.RFC3339, tms[1])
+		if err != nil {
+			log.Printf("Unable to parse time, error: %s", err)
+		}
+
+		times = append(times, startTime.Unix()*int64(milliSec))
+		times = append(times, endTime.Unix()*int64(milliSec)) // times should be in milliseconds of Unix since epoch
+		return times
+	}
+
+	for _, v := range strings.Split(trange, "-") {
+		v = strings.Trim(v, " ")
+		var t int64
+		if v == "now" {
+			t = time.Now().UTC().Unix()
+		} else if len(v) == 10 { // unix since epoch
+			value, err := strconv.Atoi(v)
+			if err != nil {
+				log.Fatalf("Unable to parse given time value: %v\n", v)
+			}
+			t = int64(value)
+		} else {
+			var value string
+			var offset int64
+			if strings.HasSuffix(v, "s") || strings.HasSuffix(v, "sec") || strings.HasSuffix(v, "secs") {
+				value = strings.Split(v, "s")[0]
+				offset = 1
+			} else if strings.HasSuffix(v, "m") || strings.HasSuffix(v, "min") || strings.HasSuffix(v, "mins") {
+				value = strings.Split(v, "m")[0]
+				offset = 60
+			} else if strings.HasSuffix(v, "h") || strings.HasSuffix(v, "hr") || strings.HasSuffix(v, "hrs") {
+				value = strings.Split(v, "h")[0]
+				offset = 60 * 60
+			} else if strings.HasSuffix(v, "d") || strings.HasSuffix(v, "day") || strings.HasSuffix(v, "days") {
+				value = strings.Split(v, "d")[0]
+				offset = 60 * 60 * 24
+			} else {
+				log.Fatalf("Unable to parse given time value: %v\n", v)
+			}
+			if v, e := strconv.Atoi(value); e == nil {
+				t = time.Now().UTC().Add(time.Duration((int64(v) * offset)) * -1 * time.Second).Unix()
+			}
+		}
+		if t == 0 {
+			log.Fatalf("Unable to parse given time value: %v\n", v)
+		}
+		times = append(times, t*milliSec) // times should be in milliseconds of Unix since epoch
+	}
+	return times
+}
+
+func main() {
+
+	flag.StringVar(&token, "token", "", "Authentication token to use")
+	flag.StringVar(&tags, "tags", "", "List of tags seperated by comma")
+	flag.StringVar(&action, "action", "", "Action to be performed, eg. delete, update etc.")
```

Comment:
do not put **eg**, be precise which actions are support. I suggest that the tool should have this list: create, update, delete, list and you should implement them. And, clearly say which is default one (which is list as far as I can tell).

Reply:
fixed